File: erp-inventory/src/main/java/com/own/erp/inventory/entity/Inventory.java
package com.own.erp.inventory.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * @author : chenyi
 * @Date : 2026/9/3
 * @Description : 库存(只能经 inventory_flow 变动)(inventory)。
 *     @Builder 仅用于 change() 新行一次成型(docs/07 §1 分级⑤);存量行是读改写生命周期,只走 setter(白名单④)
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("inventory")
public class Inventory {

    /** 主键 */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** SKU ID(product_sku.id) */
    private Long skuId;

    /** 仓库ID(warehouse.id) */
    private Long warehouseId;

    /** 在库 */
    private Integer qtyOnHand;

    /** 占用(已分配未发货) */
    private Integer qtyLocked;

    /** 在途(采购未入库) */
    private Integer qtyTransit;

    /** 可用=在库-占用,由InventoryService同事务维护,禁止旁路update */
    private Integer qtyAvailable;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}

File: erp-web/src/api/interface/inventory/inventory.ts
/**
 * 本文件由 pnpm gen:page 生成(spec 行式拍板 + tools/openapi.json 快照)
 * 默认存在即跳过:人工改动不会被 --force 之外的任何方式覆盖;重新生成前先 diff 人工改动
 * 框架代码禁手改;业务槽位一律 TODO(编号),编号已登记 TODO.md
 */

/** 库存查询接口类型(gen:page 从 openapi 快照生成;对齐后端契约,禁手抄字段) */

/** 库存查询实体(分页行 / 详情) */
export interface InventoryResponse {
  /** id */
  id: number
  /** 内部SKU */
  skuId: number
  /** 仓库ID */
  warehouseId: number
  /** 在库数量 */
  qtyOnHand: number
  /** 占用数量 */
  qtyLocked: number
  /** 在途数量 */
  qtyTransit: number
  /** 可用数量 */
  qtyAvailable: number
  /** createdAt */
  createdAt: string
  /** 更新时间 */
  updatedAt: string
}

/** 库存查询分页查询入参(不含分页参数) */
export interface InventoryQuery {
  /** 内部SKU */
  skuId?: number
  /** 仓库ID */
  warehouseId?: number
}
